src/backend/crossterm.rs

```diff
@@ -127,6 +147,245 @@ where
     }
 }
 
+impl CrosstermBackend<io::Stdout> {
+    /// Creates a new `CrosstermBackend` with `std::io::stdout`.
+    ///
+    /// # Example
+    ///
+    /// ```rust,no_run
+    /// # use ratatui::backend::CrosstermBackend;
+    /// let backend = CrosstermBackend::stdout();
+    /// ```
+    pub fn stdout() -> Self {
+        Self::new(io::stdout())
+    }
+
+    /// Creates a new `CrosstermBackend` with `std::io::stdout` and default settings.
+    ///
+    /// This enables raw mode and switches to the alternate screen. Mouse support is not enabled.
+    ///
+    /// Raw mode and alternate screen are restored when the `CrosstermBackend` is dropped.
+    ///
+    /// # Example
+    ///
+    /// ```rust,no_run
+    /// # use ratatui::backend::CrosstermBackend;
+    /// let backend = CrosstermBackend::stdout_with_defaults()?;
+    /// # std::io::Result::Ok(())
+    /// ```
+    pub fn stdout_with_defaults() -> io::Result<Self> {
+        Self::stdout().with_raw_mode()?.with_alternate_screen()
+    }
+}
+
+impl CrosstermBackend<io::Stderr> {
```

Comment:
Yes. It's used and wanted by users and is a natural thing to have sure to symmetry

Reply:
What is a common use case for this?

Reply:
https://github.com/search?q=crosstermbackend+stderr+AND+%28NOT+org%3Aratatui-org%29+AND+%28NOT+is%3Afork%29&type=code

Reply:
That’s still no real use case why that’s of interest instead of stdout. There is a blogpost comparing the two resulting in stdout being faster. And some other weird usages in there.

Normally stderr is used to not print to stdout as that’s used for piping but everything on the alternate mode doesn’t end up in a pipe?

@orhun has a gpg-tui which uses stderr instead of stdout. Curious why that is?

which might also be fun in the future with a panic handler: the panic handler needs to use the stateless reset method and supply a target. Per default stdout. What happens when stderr is used to start and the panic handler uses stdout?

Reply:
> That’s still no real use case why that’s of interest instead of stdout. There is a blogpost comparing the two resulting in stdout being faster. And some other weird usages in there.

There are 616 results. That's enough to suggest there's a reasonable use case.

> everything on the alternate mode doesn’t end up in a pipe?

This is (verifiably) incorrect

> which might also be fun in the future with a panic handler: the panic handler needs to use the stateless reset method and supply a target. Per default stdout. What happens when stderr is used to start and the panic handler uses stdout?

Stdout and Stderr don't have separate concepts of the things which are being restored. Choosing one or another is fairly arbitrary, as changes on one affect both Writers. Something that's on my someday investigate list is to look at whether there's a good pattern for choosing the tty instead of stdout/err for the writter. The termwiz backend does this automatically, which makes the output of a Ratatui app not end up in the pipe.

The following example code does exactly what you'd reasonably expect. It restores the terminal to its original state even though the setup was on stdout and the restore was on stderr. You can also check that it pipes (cargo run | wc -l returns various values that align with the number of keys you press).

```rust
use std::io::{stderr, stdout, Write};

use crossterm::{
    cursor::MoveToNextLine,
    event::{
        self, DisableBracketedPaste, DisableFocusChange, DisableMouseCapture, EnableBracketedPaste,
        EnableFocusChange, EnableMouseCapture, Event, KeyboardEnhancementFlags,
        PopKeyboardEnhancementFlags, PushKeyboardEnhancementFlags,
    },
    execute,
    terminal::{disable_raw_mode, enable_raw_mode, EnterAlternateScreen, LeaveAlternateScreen},
};

fn main() -> color_eyre::Result<()> {
    let mut stdout = stdout();
    let mut stderr = stderr();
    enable_raw_mode()?;

    execute!(
        stdout,
        EnterAlternateScreen,
        EnableMouseCapture,
        EnableBracketedPaste,
        EnableFocusChange,
        PushKeyboardEnhancementFlags(KeyboardEnhancementFlags::DISAMBIGUATE_ESCAPE_CODES)
    )?;
    stdout.flush()?;
    write!(stdout, "stdout (alt)")?;
    execute!(stdout, MoveToNextLine(1))?;
    write!(stderr, "stderr (alt)")?;
    execute!(stdout, MoveToNextLine(1))?;

    loop {
        let event = event::read()?;
        write!(stdout, "{:?}", event)?;
        execute!(stdout, MoveToNextLine(1))?;
        if let Event::Key(event) = event {
            if event.code == event::KeyCode::Char('q') {
                break;
            }
        }
        println!()
    }
    execute!(
        stderr,
        LeaveAlternateScreen,
        DisableMouseCapture,
        DisableBracketedPaste,
        DisableFocusChange,
        PopKeyboardEnhancementFlags
    )?;
    stderr.flush()?;
    disable_raw_mode()?;
    write!(stdout, "stdout (alt)")?;
    execute!(stdout, MoveToNextLine(1))?;
    write!(stderr, "stderr (alt)")?;
    execute!(stdout, MoveToNextLine(1))?;
    Ok(())
}
```